### PR TITLE
Some Pattern Matcher tests

### PR DESCRIPTION
**Pull request by reddyn12**

### CODE REVIEW

**Comment by Qazalin** [comment]
1. in some places your CONST arg doesn't match dtype
2. Do your tests need UOps.GEP or UOps.LOAD? GEP has an integer arg, UOps.LOAD doesn't have arg

**Comment by reddyn12** [comment]
Also timed pre merge: ```0.85s user 0.32s system 242% cpu 0.483 total```
Post merge: ```0.71s user 0.27s system 287% cpu 0.341 total```

File: test/test_pattern_matcher.py [hunk 1]
```diff
@@ -50,14 +53,125 @@ def test_vin_permutations(self):
...
+    self.assert_equiv_uops(matcher.rewrite(a1), c0_int)
+    self.assert_equiv_uops(matcher.rewrite(a2), c0_float)
+
+  def test_fold_neg_mul(self):
+    matcher = PatternMatcher([({"uop": UOps.ALU, "arg": BinaryOps.MUL,
+                                       "vin": [{"__name__": "x"}, {"uop": UOps.CONST, "arg": -1}]}, lambda x: -x)])
+    c1 = UOp(UOps.CONST, dtypes.float, arg=42.0)
+    c2 = UOp(UOps.CONST, dtypes.float, arg=-1.0)
+    a1 = UOp(UOps.ALU, dtypes.float, (c1,c2), BinaryOps.MUL)
+    a2 = UOp(UOps.ALU, dtypes.float, (c1), UnaryOps.NEG)
```
Comment (Qazalin) [inline test/test_pattern_matcher.py:127]: how does this pass the linter

Comment (reddyn12) [reply test/test_pattern_matcher.py]: I fixed it for this test. Did some searching online and both ruff and pylint require custom rules to catch it.

File: test/test_pattern_matcher.py [hunk 2]
```diff
@@ -50,14 +53,125 @@ def test_vin_permutations(self):
...
+
+  def test_neg_cmplt(self):
+    matcher = PatternMatcher([({"uop": UOps.ALU, "arg": BinaryOps.CMPLT, "vin": ({"uop": UOps.ALU, "arg": UnaryOps.NEG, "vin": ({"__name__": "x"},)},
+                        {"__name__": "c", "uop": UOps.CONST, "dtype": dtypes.int})},
+                        lambda c,x: UOp(UOps.ALU, dtypes.bool, (UOp.const(c.dtype, -c.arg), x), BinaryOps.CMPLT))])
+    c1 = UOp(UOps.CONST, dtypes.int, arg=42)
+    c1_neg = UOp(UOps.CONST, dtypes.int, arg=-42)
+    c2 = UOp(UOps.CONST, dtypes.int, arg=50)
+    a1 = UOp(UOps.ALU, dtypes.int, (c2,), UnaryOps.NEG)
+    a2 = UOp(UOps.ALU, dtypes.int, (a1,c1), BinaryOps.CMPLT)
```
Comment (Qazalin) [inline test/test_pattern_matcher.py:171]: ?

Comment (reddyn12) [reply test/test_pattern_matcher.py:171]: I built the tree to match the pattern, whats wrong with it?

**Comment by geohot** [comment]
There's whitespace changes mixed in with functional changes.

**Pull request closed without merging**